packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,209 @@
+import {Checkbox} from './Checkbox';
+import {FocusableRef, FocusableRefValue} from '@react-types/shared';
+import {focusRing, style} from '../style' with {type: 'macro'};
+import {getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import React, {createContext, forwardRef, ReactNode, useContext, useRef} from 'react';
+import {SelectBoxContext} from './SelectBoxGroup';
+import {useFocusableRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SelectBoxProps extends 
+  Omit<CheckboxProps & RadioProps, 'className' | 'style' | 'children'>, StyleProps {
+  /**
+   * The value of the SelectBox.
+   */
+  value: string,
+  /**
+   * The label for the element.
+   */
+  children?: ReactNode,
+  /**
+   * Whether the SelectBox is disabled.
+   */
+  isDisabled?: boolean,
+  /**
+   * Whether the SelectBox is selected (controlled).
+   */
+  isSelected?: boolean,
+  /**
+   * Handler called when the SelectBox selection changes.
+   */
+  onChange?: (isSelected: boolean) => void
+}
+
+export const SelectBoxItemContext = createContext<ContextValue<Partial<SelectBoxProps>, FocusableRefValue<HTMLLabelElement>>>(null);
+
+// Simple basic styling with proper dark mode support
+const selectBoxStyles = style({
+  ...focusRing(),
+  display: 'flex',
+  flexDirection: 'column',
+  lineHeight: 'title',
+  justifyContent: 'center',
+  flexShrink: 0,
+  alignItems: 'center',
+  fontFamily: 'sans',
+  font: 'ui',
+  
+  // Vertical orientation (default) - Fixed square dimensions
+  width: {
+    default: {
+      size: {
+        XS: 100,
+        S: 128,
+        M: 136,
+        L: 160,
+        XL: 192
+      }
+    },
+    orientation: {
+      horizontal: {
+        size: {
+          XS: 'auto',
+          S: 'auto',
+          M: 'auto',
+          L: 'auto',
+          XL: 'auto'
+        }
+      }
+    }
+  },
+  
+  height: {
+    default: {
+      size: {
+        XS: 100,
+        S: 128,
+        M: 136,
+        L: 160,
+        XL: 192
+      }
+    },
+    orientation: {
+      horizontal: {
+        size: {
+          XS: 'auto',
+          S: 'auto',
+          M: 'auto',
+          L: 'auto',
+          XL: 'auto'
+        }
+      }
+    }
+  },
+  
+  minWidth: {
+    orientation: {
+      horizontal: 160
+    }
+  },
+  
+  maxWidth: {
+    orientation: {
+      horizontal: 272
+    }
+  },
+
+  padding: {
+    size: {
+      XS: 12,
+      S: 16,
+      M: 20,
+      L: 24,
+      XL: 28
+    }
+  },
+  
+  borderRadius: 'lg',
+  backgroundColor: 'layer-2',
+  boxShadow: {
+    default: 'emphasized',
+    isHovered: 'elevated',
+    isSelected: 'elevated',
+    forcedColors: 'none'
+  },
+  position: 'relative',
+  borderWidth: 2,
+  borderStyle: {
+    default: 'solid',
+    isSelected: 'solid'
+  },
+  borderColor: {
+    default: 'transparent',
+    isSelected: 'gray-900',
+    isFocusVisible: 'transparent'
```

Comment:
i think this needs to go above `isSelected` since selection should have precedence